third_party/move/move-binary-format/src/compatibility.rs

```diff
@@ -153,30 +170,45 @@ impl Compatibility {
                 // private can become public or friend, or stay private
                 (Visibility::Private, _) => true,
             };
-            let is_entry_compatible = if old_module.file_format_version < VERSION_5
-                && new_module.file_format_version < VERSION_5
-            {
-                // if it was public(script), it must remain pubic(script)
-                // if it was not public(script), it _cannot_ become public(script)
-                old_func.is_entry == new_func.is_entry
-            } else {
-                // If it was an entry function, it must remain one.
-                // If it was not an entry function, it is allowed to become one.
-                !old_func.is_entry || new_func.is_entry
-            };
+            let is_entry_compatible =
+                if old_view.module().version < VERSION_5 && new_view.module().version < VERSION_5 {
+                    // if it was public(script), it must remain public(script)
+                    // if it was not public(script), it _cannot_ become public(script)
+                    old_func.is_entry() == new_func.is_entry()
+                } else {
+                    // If it was an entry function, it must remain one.
+                    // If it was not an entry function, it is allowed to become one.
+                    !old_func.is_entry() || new_func.is_entry()
+                };
             if !is_vis_compatible
                 || !is_entry_compatible
-                || old_func.parameters != new_func.parameters
-                || old_func.return_ != new_func.return_
-                || !fun_type_parameters_compatibile(
-                    &old_func.type_parameters,
-                    &new_func.type_parameters,
+                || !signature_compatible(
+                    old_module,
+                    old_func.parameters(),
+                    new_module,
+                    new_func.parameters(),
+                )
+                || !signature_compatible(
+                    old_module,
+                    old_func.return_(),
```

Comment:
Can we rename `return_` into `return_ty` or something? This naming with _ and not fully saying if something is a type or a name, etc. is really confusing.